specification/quantum/resource-manager/Microsoft.Quantum/preview/2019-11-04-preview/examples/quantumWorkspacesPut.json

```diff
@@ -0,0 +1,115 @@
+{
+  "parameters": {
+    "workspaceName": "quantumworkspace1",
+    "resourceGroupName": "quantumResourcegroup",
+    "subscriptionId": "1C4B2828-7D49-494F-933D-061373BE28C2",
+    "api-version": "2019-11-04-preview",
+    "QuantumWorkspace": {
+      "location": "West US",
+      "properties": {
+        "providers": [
+          {
+            "providerId": "Honeywell",
+            "providerSku": "Basic"
+          },
+          {
+            "providerId": "IonQ",
+            "providerSku": "Basic"
+          },
+          {
+            "providerId": "OneQBit",
+            "providerSku": "Basic"
+          }
+        ],
+        "storageAccount": "/subscriptions/1C4B2828-7D49-494F-933D-061373BE28C2/resourceGroups/quantumResourcegroup/providers/Microsoft.Storage/storageAccounts/testStorageAccount"
+      }
+    }
+  },
+  "responses": {
+    "200": {
+      "body": {
+        "name": "quantumworkspace1",
+        "properties": {
+          "providers": [
+            {
+              "providerId": "Honeywell",
+              "providerSku": "Basic",
+              "applicationName": "quantumworkspace1-h1",
```

Comment:
this property is present in the response, but not the request. Is that because it is read-only or optional?

Reply:
No, it's because the RP will create asynchronously a managed app for each of the requested providers. These fields are populated by the RP based on the managed-app created and associated for the corresponding workspace.

Reply:
so they are readonly, meaning the customer cannot change/specify these values right?